[CI] CoreWithSecurityClientYamlTestSuiteIT test {yaml=search.retrievers/result-diversification/10_mmr_result_diversification_retriever/Test MMR result diversification multiple indexes} failing

**Build Scans:**
- [elasticsearch-intake #34576 / part4](https://gradle-enterprise.elastic.co/s/7lkvqjmye5t7m)
- [elasticsearch-pull-request #120200 / part-4](https://gradle-enterprise.elastic.co/s/v6i7zwgwsilnu)
- [elasticsearch-periodic-platform-support #11838 / windows-2025_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/75tfohxdbakpc)
- [elasticsearch-periodic-platform-support #11838 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/blzeqvfeete34)
- [elasticsearch-periodic-java-ea #171 / checkpart4_windows-2025](https://gradle-enterprise.elastic.co/s/yh7dsladxsgto)
- [elasticsearch-periodic-platform-support #11834 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/o72ciozj3wv4m)
- [elasticsearch-periodic-platform-support #11830 / windows-2025_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/hqk6nqcvyli5u)
- [elasticsearch-periodic-platform-support #11830 / windows-2022_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/hrt5padzz6gga)
- [elasticsearch-periodic-java-ea #169 / checkpart4_windows-2025](https://gradle-enterprise.elastic.co/s/zbdp53f6b3h2y)
- [elasticsearch-periodic-platform-support #11826 / windows-2025_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/fznwqrzjurxlm)

**Reproduction Line:**
```
./gradlew ":x-pack:qa:core-rest-tests-with-security:yamlRestTest" --tests "org.elasticsearch.xpack.security.CoreWithSecurityClientYamlTestSuiteIT" -Dtests.method="test {yaml=search.retrievers/result-diversification/10_mmr_result_diversification_retriever/Test MMR result diversification multiple indexes}" -Dtests.seed=8CE839E0BA661A72 -Dtests.locale=yue-CN -Dtests.timezone=NZ -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.security.CoreWithSecurityClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Byaml%3Dsearch.retrievers%2Fresult-diversification%2F10_mmr_result_diversification_retriever%2FTest%20MMR%20result%20diversification%20multiple%20indexes%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.lang.AssertionError: 
Expected: "second text other"
     but: was "second text"
```

**Issue Reasons:**
- [main] 11 failures in test test {yaml=search.retrievers/result-diversification/10_mmr_result_diversification_retriever/Test MMR result diversification multiple indexes} (1.1% fail rate in 974 executions)
- [main] 3 failures in step windows-2025_checkpart4_platform-support-windows (50.0% fail rate in 6 executions)
- [main] 4 failures in step windows-2022_checkpart4_platform-support-windows (57.1% fail rate in 7 executions)
- [main] 2 failures in step checkpart4_windows-2025 (50.0% fail rate in 4 executions)
- [main] 4 failures in pipeline elasticsearch-periodic-platform-support (57.1% fail rate in 7 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-java-ea (33.3% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -401,9 +401,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.parser.promql.PromqlParserTests
   method: testInstantVectorExpected
   issue: https://github.com/elastic/elasticsearch/issues/142420
-- class: org.elasticsearch.xpack.security.CoreWithSecurityClientYamlTestSuiteIT
-  method: test {yaml=search.retrievers/result-diversification/10_mmr_result_diversification_retriever/Test MMR result diversification multiple indexes}
-  issue: https://github.com/elastic/elasticsearch/issues/142422
 - class: org.elasticsearch.indices.stats.IndexStatsIT
   method: testFieldDataStats
   issue: https://github.com/elastic/elasticsearch/issues/142424

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.retrievers/result-diversification/10_mmr_result_diversification_retriever.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.retrievers/result-diversification/10_mmr_result_diversification_retriever.yml
@@ -442,25 +442,24 @@ teardown:
         index: "test-result-diversification-index*"
         body:
           retriever:
-            standard:
-              query:
-                match_all: { }
-              sort:
-                id:
-                  order: asc
+            knn:
+              field: "textvector"
+              query_vector: [ 0.5, 0.2, 0.4, 0.4 ]
+              k: 10
+              num_candidates: 10
 
-  - match: { hits.total.value: 19 }
+  - match: { hits.total.value: 10 }
   - length: { hits.hits: 10 }
-  - match: { hits.hits.0._source.textbody: "first text" }
-  - match: { hits.hits.1._source.textbody: "second text" }
-  - match: { hits.hits.2._source.textbody: "second text duplicate" }
-  - match: { hits.hits.3._source.textbody: "third text" }
-  - match: { hits.hits.4._source.textbody: "third text duplicate" }
-  - match: { hits.hits.5._source.textbody: "fourth text" }
-  - match: { hits.hits.6._source.textbody: "fifth text" }
-  - match: { hits.hits.7._source.textbody: "sixth text" }
-  - match: { hits.hits.8._source.textbody: "seventh text" }
-  - match: { hits.hits.9._source.textbody: "eighth text" }
+  - match: { hits.hits.0._source.textbody: "second text" }
+  - match: { hits.hits.1._source.textbody: "second text duplicate" }
+  - match: { hits.hits.2._source.textbody: "third text" }
+  - match: { hits.hits.3._source.textbody: "third text duplicate" }
+  - match: { hits.hits.4._source.textbody: "first text" }
+  - match: { hits.hits.5._source.textbody: "sixth text" }
+  - match: { hits.hits.6._source.textbody: "ninth text" }
+  - match: { hits.hits.7._source.textbody: "third text other" }
+  - match: { hits.hits.8._source.textbody: "third text duplicate other" }
+  - match: { hits.hits.9._source.textbody: "fifth text other" }
 
   - do:
       search:

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.retrievers/result-diversification/10_mmr_result_diversification_retriever.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.retrievers/result-diversification/10_mmr_result_diversification_retriever.yml
@@ -118,22 +118,22 @@ setup:
         index: test-result-diversification-index-other
         refresh: true
         body: |
-          {"index":{}}
-          {"id": 12, "textbody": "first text other", "textvector": [0.4, 0.2, 0.4, 0.4], "keywordfield": "test1"}
-          {"index":{}}
-          {"id": 23, "textbody": "second text other", "textvector": [0.4, 0.2, 0.3, 0.3], "keywordfield": "test2"}
-          {"index":{}}
-          {"id": 14, "textbody": "second text duplicate other", "textvector": [0.4, 0.2, 0.3, 0.3], "keywordfield": "test2_dup"}
-          {"index":{}}
-          {"id": 15, "textbody": "third text other", "textvector": [0.4, 0.1, 0.3, 0.3], "keywordfield": "test3"}
-          {"index":{}}
-          {"id": 16, "textbody": "third text duplicate other", "textvector": [0.4, 0.1, 0.3, 0.3], "keywordfield": "test3_dup"}
-          {"index":{}}
-          {"id": 17, "textbody": "fourth text other", "textvector": [0.1, 0.9, 0.5, 0.9], "keywordfield": "test4"}
-          {"index":{}}
-          {"id": 18, "textbody": "fifth text other", "textvector": [0.1, 0.9, 0.5, 0.9], "keywordfield": "test5"}
-          {"index":{}}
-          {"id": 19, "textbody": "sixth text other", "textvector": [0.05, 0.05, 0.05, 0.05], "keywordfield": "test6"}
+          { "index": { } }
+          { "id": 12, "textbody": "first text other", "textvector": [ 0.14, 0.5, 0.49, 0.36 ], "keywordfield": "test1" }
+          { "index": { } }
+          { "id": 23, "textbody": "second text other", "textvector": [ 0.63, 0.63, 0.51, 0.07 ], "keywordfield": "test2" }
+          { "index": { } }
+          { "id": 14, "textbody": "second text duplicate other", "textvector": [ 0.63, 0.63, 0.51, 0.07 ], "keywordfield": "test2_dup" }
+          { "index": { } }
+          { "id": 15, "textbody": "third text other", "textvector": [ 0.3, 0.24, 0.97, 0.82 ], "keywordfield": "test3" }
+          { "index": { } }
+          { "id": 16, "textbody": "third text duplicate other", "textvector": [ 0.3, 0.24, 0.97, 0.82 ], "keywordfield": "test3_dup" }
+          { "index": { } }
+          { "id": 17, "textbody": "fourth text other", "textvector": [ 0.88, 0.82, 0.96, 0.06 ], "keywordfield": "test4" }
+          { "index": { } }
+          { "id": 18, "textbody": "fifth text other", "textvector": [ 0.36, 0.03, 0.1, 0.62 ], "keywordfield": "test5" }
+          { "index": { } }
+          { "id": 19, "textbody": "sixth text other", "textvector": [ 0.19, 0.44, 0.68, 0.35 ], "keywordfield": "test6" }
 
   - do:
       headers:
@@ -481,9 +481,9 @@ teardown:
 
   - match: { hits.total.value: 10 }
   - length: { hits.hits: 3 }
-  - match: { hits.hits.0._source.textbody: "second text other" }
-  - match: { hits.hits.1._source.textbody: "third text other" }
-  - match: { hits.hits.2._source.textbody: "third text duplicate other" }
+  - match: { hits.hits.0._source.textbody: "second text" }
+  - match: { hits.hits.1._source.textbody: "sixth text" }
+  - match: { hits.hits.2._source.textbody: "ninth text" }
 
 ---
 "Test MMR result diversification with default size should return results":

